com.unity.render-pipelines.core/Tests/Editor/UpgradeUtility_UnitTests.cs

```diff
@@ -0,0 +1,232 @@
+using System.Collections.Generic;
+using System.Linq;
+using Moq;
+using NUnit.Framework;
+using IMaterial = UnityEditor.Rendering.UpgradeUtility.IMaterial;
+using RenameType = UnityEditor.Rendering.MaterialUpgrader.RenameType;
+using static UnityEditor.Rendering.Tests.UpgraderTestUtility;
+
+namespace UnityEditor.Rendering.Tests
```

Comment:
Graphics: These tests cover the internal UpgradeUtility class that I propose you should add/adjust into public API for your upgrader tool